Lib/_pyrepl/unix_console.py

```diff
@@ -325,8 +328,11 @@ def prepare(self):
         """
         Prepare the console for input/output operations.
         """
-        self.__svtermstate = tcgetattr(self.input_fd)
-        raw = self.__svtermstate.copy()
+        # gh-130168: prevents signal handlers from overwriting the original state
+        if not hasattr(self, '_UnixConsole__svtermstate'):
+            self.__svtermstate = tcgetattr(self.input_fd)
+
```

Comment:
```suggestion
        if self.__svtermstate is None:
            self.__svtermstate = tcgetattr(self.input_fd)
```

And initialize the attribute in the constructor to None.